components/coreapi/utxo.go

```diff
@@ -0,0 +1,43 @@
+package coreapi
+
+import (
+	"github.com/iotaledger/inx-app/pkg/httpserver"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/ledgerstate"
+	restapipkg "github.com/iotaledger/iota-core/pkg/restapi"
+	"github.com/labstack/echo/v4"
+)
+
+func getOutput(c echo.Context) (*ledgerstate.Output, error) {
+	outputID, err := httpserver.ParseOutputIDParam(c, restapipkg.ParameterOutputID)
+	if err != nil {
+		return nil, err
+	}
+
+	output, err := deps.Protocol.MainEngineInstance().Ledger.Output(outputID)
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func getOutputMetadata(c echo.Context) (*outputMetadataResponse, error) {
+	output, err := getOutput(c)
+	if err != nil {
+		return nil, err
+	}
+	outputID := output.OutputID()
+	slotCommitment := deps.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment()
+	spent, err := deps.Protocol.MainEngineInstance().Ledger.IsOutputSpent(outputID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &outputMetadataResponse{
+		BlockID:            output.BlockID().ToHex(),
+		TransactionID:      outputID.TransactionID().ToHex(),
+		OutputIndex:        outputID.Index(),
+		IsSpent:            spent,
```

Comment:
In HORNET we had some extra optional fields when the output was spent, so that a client could "follow" the funds

Reply:
We also have a few extra fields in the OutputMetadata, I'll add a TODO here.